lib/awful/rules.lua

```diff
@@ -619,7 +514,9 @@ end
 -- @client c The client.
 -- @tab props Properties to apply.
 -- @tab[opt] callbacks Callbacks to apply.
-function rules.execute(c, props, callbacks)
+-- @function awful.rules.execute
+
+crules.execute = function(_, c, props, callbacks)
```

Comment:
Should be moved also to the new module?!

Reply:
It is in `gears.matcher` too. This is a "virtual function override" of `gears.matcher.execute`. I kept it this way because `awful.rules` in Awesome 4.x still uses hardcoded prioritt groups instead of using the graph dependency solver. That code makes zero sense in `gears.matcher` and its features fully redundant with the add/remove_source methods.